nighthawk/source/client/output_formatter_impl.h

```diff
@@ -0,0 +1,54 @@
+#pragma once
+
+#include "nighthawk/client/output_formatter.h"
+
+#include "envoy/common/time.h"
+
+#include "common/protobuf/protobuf.h"
+
+#include <cstdint>
+
+namespace Nighthawk {
+namespace Client {
+
+class OutputFormatterImpl : public OutputFormatter {
+public:
+  OutputFormatterImpl(Envoy::TimeSource& time_source, const Options& options,
+                      const std::vector<StatisticPtr>& merged_statistics,
+                      const std::map<std::string, uint64_t>& merged_counters);
+
+  Envoy::ProtobufTypes::MessagePtr toProto() const;
```

Comment:
This is public, but not part of the interface, can you elaborate on the reasoning?

Reply:
My bad, I changed it to be `protected`; there's been no need to expose it further at the moment. Lets defer that to if the need for doing so arises? (When that happens, we may want to relocate the `toProto` to it's own layer ?)